src/app/hooks/flights-state.ts

```diff
@@ -0,0 +1,53 @@
+import { useEffect, useState } from "react";
+import { DropdownItem } from "../models/dropdown-item";
+import { FlightParameters, FlightSearchInfo, FlightsStateProps, FlightTravellersParameters, TravelType } from "../models/flight-parameters";
+import { AirportCodes } from "../models/airport-codes.enum";
+import { getFlights } from "../services/flights-search-service";
+
+export default function useFlightsState(): FlightsStateProps {
+    const [items, setItems] = useState<DropdownItem<string>[]>([]);
+    const [isLoading, setIsLoading] = useState(false);
+    const [parameters, setParameters] = useState<FlightParameters>({
+        from: '',
+        to: '',
+        travelType: TravelType.OneWay,
+        departureDate: new Date(),
+        adults: 1
+    });
+    const [flights, setFlights] = useState<FlightSearchInfo[]>([]);
+
+    useEffect(() => {
+        setItems([
+            { label: 'Chennai', value: AirportCodes.Chennai },
+            { label: 'Jaipur', value: AirportCodes.Jaipur }
+        ]);
+    }, []);
+
+    const setFrom = (from: string) => setParameters((current) => ({ ...current, from }));
```

Comment:
This code can be optimized as we discussed in yesterday's call

Reply:
the components that I am using do not send the event and send the value directly from the event (similar will be the case with external components), also the name field is not exposed
But I can still reduce the number of exported methods by doing below if you'd like

`const setBasicFilterData = (data: { [k in keyof FlightParameters]: FlightParameters[k] }) => setParameters((current) => ({ ...current, ...data }));`

and consuming it like 
`<CustomElement valueChange={(value) => setBasicFilterData({ key: value })}`